backend/layers/persistence/persistence_interface.py

```diff
@@ -0,0 +1,223 @@
+from datetime import datetime
+from typing import List, Optional, Iterable
+from backend.layers.common.entities import (
+    CollectionId,
+    CollectionMetadata,
+    CollectionVersion,
+    CollectionVersionId,
+    DatasetArtifact,
+    DatasetArtifactId,
+    DatasetConversionStatus,
+    DatasetId,
+    DatasetMetadata,
+    DatasetProcessingStatus,
+    DatasetStatus,
+    DatasetUploadStatus,
+    DatasetValidationStatus,
+    DatasetVersion,
+    DatasetVersionId,
+)
+
+
+class DatabaseProviderInterface:
+    def create_canonical_collection(self, owner: str, collection_metadata: CollectionMetadata) -> CollectionVersion:
+        """
+        Creates a new canonical collection, generating a canonical collection_id and a new version_id.
+        Returns the newly created CollectionVersion
+        """
+        pass
+
+    def get_collection_version(self, version_id: CollectionVersionId) -> CollectionVersion:
```

Comment:
Sorry, not entirely sure what you mean. We don't want to combine the two because we want to support being able to retrieve an arbitrary CollectionVersion, even if it's not canonical, by just the version ID. And we want to avoid JOINs because we're observing high CPU usage on our DBs as-is and want to be as NoSQL-like as possible so we can more easily transition to a NoSQL persistence solution in the future. 

Apologies if that's not what you meant tho